### PR TITLE
ACS-8149: Adding HxI API call to answers endpoint.

### PR DESCRIPTION
**Pull request by mpichura**
As per https://hyland.atlassian.net/browse/ACS-8150 - the result is not yet returned by the REST API but only logged (at DEBUG level).

### CODE REVIEW

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClientTest.java [hunk 1]
```diff
@@ -114,4 +119,42 @@ void shouldThrowOnNotExpectedStatusCode()
                 new Question("Who won last year's Super Bowl?", "")));
     }
 
+    @SneakyThrows
+    @Test
+    void shouldReturnAnswer()
+    {
+        // given
+        String questionId = "dummy-id-1234";
+        String answer = "The Kansas City Chiefs won last year's Super Bowl.";
```
Comment (tpage-alfresco) [inline hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClientTest.java:128]: Good knowledge. This is going to be hard to keep up to date :upside_down_face:

**Review by tpage-alfresco — APPROVED** [review]
LGTM :+1:

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java [hunk 2]
```diff
@@ -105,4 +106,27 @@ public String askQuestion(Question question)
...
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(config.getAnswerUrl().formatted(questionId)))
+                    .headers(authService.getAuthHeaders())
+                    .GET()
+                    .build();
+
+            HttpResponse<String> httpResponse = client.send(request, BodyHandlers.ofString());
+            log.atDebug().log("Question with id {} received a following answer {}", questionId, httpResponse.body());
+
+            throwExceptionOnUnexpectedStatusCode(httpResponse.statusCode(), SC_OK);
```
Comment (krdabrowski) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java:123]: What will be the final server's response status code for thrown let say `EndpointClientErrorException`? `500` or `httpResponse.statusCode()`?

Comment (Pawel-608) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java:123]: we'll have to chage this part anyway to something like this:
image

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelation.java [hunk 3]
```diff
@@ -40,11 +41,12 @@
...
 {
+    private final HxInsightClient hxInsightClient;
 
     @WebApiDescription(title = "Get answers to a question")
     public CollectionWithPagingInfo<AnswerModel> readAll(String questionId, Parameters parameters)
     {
-        AnswerModel answerModel = new AnswerModel("Dummy answer");
-        return CollectionWithPagingInfo.asPagedCollection(answerModel);
+        hxInsightClient.getAnswer(questionId);
+        return CollectionWithPagingInfo.asPagedCollection(new AnswerModel("Dummy answer"));
```
Comment (Pawel-608) [inline hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelation.java:50]: why don't you pass answer to client?

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    void shouldCallHxInsightClientGetAnswer()
+    {
+        // given
+        String questionId = "questionId";
+
+        // when
+        objectUnderTest.readAll(questionId, null);
+
+        // then
+        then(mockHxInsightClient).should().getAnswer(questionId);
```
Comment (Pawel-608) [inline hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java:59]: maybe you should also test if answer is passed to the caller?

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelation.java (hunk 3 above)
Comment (mpichura) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelation.java:50]: I'll do that when I work on the next task in the story. Will also add some more tests (including integration test I hope).

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java (hunk 4 above)
Comment (mpichura) [reply hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java:59]: But it isn't yet. I'll do that when I work on the next task in the story. Will also add some more tests (including integration test I hope).

File: hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java (hunk 2 above)
Comment (mpichura) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java:123]: Will do

Comment (mpichura) [reply hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java:123]: As discussed - that will be done in a separate PR globally.

**Review by Pawel-608 — APPROVED** [review]

**Pull request merged**